src/common/__tests__/kube-helpers.test.ts

```diff
@@ -0,0 +1,179 @@
+const logger = {
+  silly: jest.fn(),
+  debug: jest.fn(),
+  log: jest.fn(),
+  info: jest.fn(),
+  error: jest.fn(),
+  crit: jest.fn(),
+};
+
+jest.mock("winston", () => ({
+  format: {
+    colorize: jest.fn(),
+    combine: jest.fn(),
+    simple: jest.fn(),
+    label: jest.fn(),
+    timestamp: jest.fn(),
+    printf: jest.fn()
+  },
+  createLogger: jest.fn().mockReturnValue(logger),
+  transports: {
+    Console: jest.fn(),
+    File: jest.fn(),
+  }
+}))
+
+interface kubeconfig {
+  apiVersion: string,
+  clusters: [{
+    name: string,
+    cluster: {
+      server: string
+    }
+  }],
+  contexts: [{
+    context: {
+      cluster: string,
+      user: string,
+    },
+    name: string
+  }],
+  users: [{
+    name: string
+  }],
+  kind: string,
+  'current-context': string,
+  preferences: {}
+}
+
+import { loadConfig } from "../kube-helpers"
+import { KubeConfig } from "@kubernetes/client-node";
+
+console.log("") // fix bug
+let mockKubeConfig: kubeconfig;
+
+describe("pre-validate context object in kubeconfig tests", () => {
+  beforeEach(() => {
+    jest.clearAllMocks()
+  });
+
+  describe('Check logger.error() output', () => {
+    it("invalid yaml string", () => {
+      //expect(loadConfig("fancy foo config")).toEqual({"clusters": [], "contexts": [], "currentContext": undefined, "users": []});
+      loadConfig("fancy foo config")
+      expect(logger.error).toHaveBeenCalledWith("Could not pre-validate contexts key in supplied config string - \"fancy foo config\". Error - TypeError: Cannot use 'in' operator to search for 'contexts' in fancy foo config");
```

Comment:
I this this is a bit too strong since detecting the whole error means any change to the error message means changing this test. I would recommend changing something else. To be far, we also don't want to propagate the typeerror up to the user. So both this check should be changed to `toInclude` and we should intercept the error and strip out any type errors.

Reply:
@Nokel81 - Updated and pushed. Could you have one quick last look to make sure all is good. Thanks.